drivers/crypto/crypto_mchp_xec_symcr.c

```diff
@@ -0,0 +1,571 @@
+{
+	if (!c) {
+		return -EINVAL;
+	}
+
+	if (mec172x_rom_hash_wait(c) != 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Called by application for update(finish==false)
+ * and compute final hash digest(finish==true)
+ */
+static int xec_symcr_do_hash(struct hash_ctx *ctx, struct hash_pkt *pkt, bool finish)
+{
+	struct xec_symcr_hash_session *hs = NULL;
+	struct mchphash *c = NULL;
+	struct mchphashstate *cstate = NULL;
+	size_t fill_len = 0, rem_len = 0;
+	int ret = 0;
+
+	if (!ctx || !pkt) {
+		return -EINVAL;
+	}
+
+	hs = (struct xec_symcr_hash_session *)ctx->drv_sessn_state;
+	c = &hs->mhctx;
+	cstate = &hs->mhstate;
+
+	if (!hs->open) {
+		LOG_ERR("Session not open");
+		return -EIO;
+	}
+
+	if (!finish && !pkt->in_len) {
+		return 0; /* nothing to do */
+	}
+
+	/* Not final digest computation and not enough data to run engine */
+	if (!finish && ((hs->blklen + pkt->in_len) < hs->blksz)) {
+		memcpy(&hs->blockbuf[hs->blklen], pkt->in_buf, pkt->in_len);
+		hs->blklen += pkt->in_len;
+		return 0;
+	}
+
+	ret = init_rom_hash_context(hs->rom_algo, c);
+	if (ret) {
+		LOG_ERR("ROM context init error %d", ret);
+		return ret;
+	}
+	ret = mchp_xec_rom_hash_resume_state_wrapper(c, cstate);
+	if (ret) {
+		LOG_ERR("Resume state error %d", ret);
+		return ret;
+	}
+
+	fill_len = pkt->in_len;
+	rem_len = 0;
+	if (!finish) {
+		rem_len = pkt->in_len & (hs->blksz - 1u);
+		fill_len = pkt->in_len & ~(hs->blksz - 1u);
+		if (hs->blklen) {
+			fill_len = ((hs->blklen + pkt->in_len) & ~(hs->blksz - 1u)) - hs->blklen;
+			rem_len = pkt->in_len - fill_len;
+		}
+	}
+
+	if (hs->blklen) {
+		ret = mchp_xec_rom_hash_feed_wrapper(c, (const uint8_t *)hs->blockbuf, hs->blklen);
+		if (ret) {
+			LOG_ERR("ROM hash feed error %d", ret);
+			return ret;
+		}
+		hs->blklen = 0; /* consumed */
+	}
+
+	ret = mchp_xec_rom_hash_feed_wrapper(c, (const uint8_t *)pkt->in_buf, fill_len);
+	if (ret) {
+		LOG_ERR("ROM hash feed error %d", ret);
+		return ret;
+	}
+
+	if (finish) {
+		ret = mchp_xec_rom_hash_digest_wrapper(c, pkt->out_buf);
+		if (ret) {
+			LOG_ERR("ROM Hash final error %d", ret);
+			return ret;
+		}
+	} else {
+		ret = mchp_xec_rom_hash_save_state(c);
+		if (ret) {
+			LOG_ERR("ROM hash save state error %d", ret);
+			return ret;
+		}
+	}
+	ret = mchp_xec_rom_hash_wait_wrapper(c);
+	if (ret) {
+		LOG_ERR("ROM hash wait error %d", ret);
+		return ret;
+	}
+	if (finish) {
+		hs->blklen = 0;
+	} else {
+		memcpy(hs->blockbuf, &pkt->in_buf[fill_len], rem_len);
+		hs->blklen = rem_len;
+	}
+
+	return 0;
+}
+
+static int xec_symcr_hash_session_begin(const struct device *dev, struct hash_ctx *ctx,
+					enum hash_algo algo)
+{
+	struct xec_symcr_data *data = dev->data;
+	struct xec_symcr_hash_session *hs = NULL;
+	struct mchphash *c = NULL;
+	struct mchphashstate *cstate = NULL;
+	enum mchp_rom_hash_alg_id rom_algo = MCHP_ROM_HASH_ALG_NONE;
+	int session_idx = 0;
+	int ret = 0;
+
+	if (!ctx) {
+		LOG_ERR("Bad parameters");
+		return -EINVAL;
+	}
```

Comment:
The API has already assumed that this is !NULL.

Reply:
Logic removed to check against !NULL.